doc/sphinx/language/gallina-extensions.rst

```diff
@@ -23,15 +23,19 @@ expressions. In this sense, the :cmd:`Record` construction allows defining
 .. cmd:: {| Record | Structure } @inductive_definition {* with @inductive_definition }
    :name: Record; Structure
 
-   The first identifier :token:`ident` is the name of the defined record and :token:`sort` is its
-   type. The optional identifier following ``:=`` is the name of its constructor. If it is omitted,
-   the default name :n:`Build_@ident`, where :token:`ident` is the record name, is used. If :token:`sort` is
-   omitted, the default sort is :math:`\Type`. The identifiers inside the brackets are the names of
-   fields. For a given field :token:`ident`, its type is :n:`forall {* @binder }, @type`.
-   Notice that the type of a particular identifier may depend on a previously-given identifier. Thus the
-   order of the fields is important. The record can depend as a whole on parameters :token:`binders`
-   and each field can also depend on its own :token:`binders`. Finally, notations can be attached to
-   fields using the :n:`decl_notations` annotation.
+   Each :n:`@inductive_definition` defines a record named by the first :n:`@ident` within.
+   The constructor name is the :n:`@ident` in the :n:`{? @ident } %{ {+; @record_field } %}` clause.
+   If the constructor name is not specified, then the default name :n:`Build_@ident` is used,
+   where :n:`@ident` is the record name.
+
+   If :n:`@type` (which must be a :n:`sort`??) is
+   omitted, the default sort is :math:`\Type`. The identifiers inside the brackets are the field names.
+   The type of each field :n:`@ident`, its type is :n:`forall {* @binder }, @type`.
+   Notice that the type of an identifier can depend on a previously-given identifier. Thus the
+   order of the fields is important. :n:`@binder` parameters may be applied to the record as a whole
+   or to individual fields.
```

Comment:
I feel like there could be fewer references to specific nonterminals that are not shown here and some generic explanations instead. But anyway, do not spend too much time making this perfect, since this is going to be moved closer to the other inductive definitions in the refactoring, and therefore, it should be easier to explain everything in order.

Reply:
Well, if you're trying to figure out the syntax for Record, you have to look at the definition of `inductive_definition` and some of the others.  It's important to be specific (there are multiple idents in the syntax).  At some points in the doc, I'm tempted to repeat some of the syntax just for easy reading (would require a small enhancement).  I think an astute reader would open the hyperlink in a new window so that they can see both the syntax and the text.

Alternatively and perhaps better, we could give a short example that the text could refer to.  If you give me an example covering the key points, I'll put it and adjust the text.  Other than that, not sure what I'd change.

What about ```(which must be a :n:`sort`??)```.

Reply:
> What about ```(which must be a :n:`sort`??)```.  

Ah, you covered this point above (it's not limited to a sort).

Reply:
> But should I say `` the default sort is :math:`\Type`. `` or `` the default type is :math:`\Type`. ``?

Probably the latter.

>  At some points in the doc, I'm tempted to repeat some of the syntax just for easy reading

I think that the planned refactoring will help and you won't have to open several pages. It doesn't make sense that records are defined that late in the manual.